.github/workflows/pull-request.yml

```
@@ -4,137 +4,400 @@
-        uses: actions/cache@v3
-        env:
-          cache-name: cache-node-modules
+      - name: Checkout this repository
+        uses: actions/checkout@v4.2.2
+      - name: Build talawa api non production environment docker image
+        run: docker buildx build --file ./docker/api.Containerfile --tag talawa_api --target non_production ./
+      - name: Check type errors
+        run: docker container run talawa_api pnpm check_code_quality
+      - name: Check if the source and target branches are different
+        if: ${{ github.event.pull_request.base.ref == github.event.pull_request.head.ref }}
+        run: |
```

actionlint (1.7.7)

"github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions for more details

(expression)